### ISSUE TITLE
RequestGetAutocompletion doesn't seem to return local-scope completions

### ISSUE THREAD

**Issue by lukehoersten** [drwebb/company-stack-ide#1]
When running the plugin, it correctly calls the stack-ide and returns a suggestion though it's missing a really obvious local scope suggestion (and in fact the right suggestion).

The suggestion it's returning doesn't fit the type either, just the letters.

Is this definitely the right call that should be used?

**Comment by drwebb** [comment]
Thanks, can confirm it's not working. Stack-ide has a few options for specifying the module to look in for auto completion, I would need to see if we can pull local scope bindings in addition. Reloading the file in stack-mode didn't help when I tested.

Currently there is no type-checking, that would be a future enhancement, right now it's doing the simplest thing that pulls in candidates from the backend.

**Comment by lukehoersten** [comment]
Does ide-backend not support type-based autocomplete or that's something not exposed in the stack-ide API?

**Comment by drwebb** [comment]
These are the two general reference points I was looking at. I don't know jack about the internals of ide-backend, so I can just comment on the exposed API I tracked down in the source.

https://github.com/commercialhaskell/stack-ide/blob/master/stack-ide-api/src/Stack/Ide/JsonAPI.hs
https://github.com/fpco/ide-backend/blob/master/ide-backend-common/IdeSession/Types/Public.hs

To answer your Q, look at the type of the auto completion request. https://github.com/commercialhaskell/stack-ide/blob/7e93bd3e03502beafb4613b7bc690fb9f0db1314/stack-ide-api/src/Stack/Ide/JsonAPI.hs#L62

It takes a source file and a prefix to complete, so no it's not exposed. I thought you thinking of some sort of post stack-ide response filtering.

**Comment by lukehoersten** [comment]
Yeah I bet you're right for the type-based completion stuff (re our other discussion)
